client/data/timeline/hooks.ts

```diff
@@ -0,0 +1,66 @@
+/** @format */
+/**
+ * External dependencies
+ */
+import { useSelect } from '@wordpress/data';
+import { STORE_NAME } from '../constants';
+
+/**
+ * Internal dependencies
+ */
+import { Timeline } from './types';
+import { Charge } from '../charges/types';
+
+export const useTimeline = (
+	chargeId: string
+): {
+	timeline?: Timeline;
+	timelineError?: Error;
+	isLoading: boolean;
+} =>
+	useSelect(
+		( select ) => {
+			const {
+				getTimeline,
+				getTimelineError,
+				isResolving,
+				getCharge,
+				getChargeError,
+			} = select( STORE_NAME );
+
+			// Make sure the charge is loaded first to get the intention ID.
+			const isChargeLoading = isResolving< boolean >( 'getCharge', [
+				chargeId,
+			] );
+			const chargeError = getChargeError< Error | undefined >( chargeId );
```

Comment:
Looking at those `Type | undefined` I want to add something like `Maybe<T> = T | undefined`  and reuse it

Reply:
I understand that reaction, but I think in this case the `Type | undefined` is clearer. I think using a `Maybe<T>` type will result in people thinking there's more behind it, e.g. a maybe monad (common in functional languages) which comes with its own interface.

I think using a `Maybe<T>` type will just end up causing confusion, so I'm hesitant to recommending that 😬

Reply:
Is there a chance to use better naming than `Mabye<T>`, e.g. `Optional<T>`, `UndefinedOr<T>`, etc. it feels weird to me carry all those `undefined`s.

Reply:
I honestly don't see much value in creating a wrapper type like that. It's just a different way to do the same thing and doesn't benefit us much IMO.

Besides, I'm pretty sure the TS community generally uses variations of `<type> | undefined` (e.g. `string | undefined`) instead of creating wrapper types.

Reply:
For this specific case `getChargeError< Error | undefined >( /* ... */ );` isn't all that different from `getChargeError< UndefinedOr< Error > >( /* ... */ )` and is really just slightly different noise 🙂